crates/wasmtime/src/compiler.rs

```diff
@@ -396,14 +501,19 @@ impl<'a> CompileInputs<'a> {
             .values()
             .all(|funcs| is_sorted_by_key(funcs, |x| x.key)));
 
-        Ok(UnlinkedCompileOutputs { outputs })
+        Ok(UnlinkedCompileOutputs {
+            outputs,
+            resource_drop_wasm_to_native_trampoline: self.resource_drop_wasm_to_native_trampoline,
+        })
     }
 }
 
 #[derive(Default)]
 pub struct UnlinkedCompileOutputs {
     // A map from kind to `CompileOutput`.
     outputs: BTreeMap<u32, Vec<CompileOutput>>,
+
+    resource_drop_wasm_to_native_trampoline: Option<CompileKey>,
```

Comment:
Kind of annoying we have to thread this through everything specially, when we worked so hard to avoid that elsewhere. Can we perhaps give it it a known compile key or something, so we can just look it up in the various output maps or whatever? Rather than add a special field to all of our compiler's structs. Not sure this is feasible but I'd like to just push in that direction if possible.